client/components/date-picker/index.jsx

```diff
@@ -120,7 +121,7 @@ class DatePicker extends PureComponent {
 			},
 		};
 
-		return merge( locale, this.props.locale );
+		return merge( {}, utils, localeUtils );
```

Comment:
I think merge might be overkill here,
```js
return {
  ...utils,
  ...localUtils,
}
```
might do the trick and be ever so slightly more performant, but let's not worry about it just now :)

Reply:
I agree but as you say I'm going to leave "as is" for now as I cannot ask people to do yet another review. This component seems very prone to error! :)